src/server/bitops_family_test.cc

```diff
@@ -420,4 +420,115 @@ TEST_F(BitOpsFamilyTest, BitOpsNot) {
+  EXPECT_EQ(1, CheckedInt({"bitpos", "a", "0", "1", "-2", "BIT"}));
+  EXPECT_EQ(38, CheckedInt({"bitpos", "a", "0", "-2", "-1", "BIT"}));
+
+  // Find set bits
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "1"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "2"}));
+  EXPECT_EQ(24, CheckedInt({"bitpos", "a", "1", "3"}));
+  EXPECT_EQ(32, CheckedInt({"bitpos", "a", "1", "4"}));
+  EXPECT_EQ(32, CheckedInt({"bitpos", "a", "1", "-1"}));
+  EXPECT_EQ(24, CheckedInt({"bitpos", "a", "1", "-2"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "-3"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "-4"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "-5"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "-6"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "-100"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "0", "0"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "0", "1"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0", "3"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0", "100"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "2", "2"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "2", "3"}));
+  EXPECT_EQ(32, CheckedInt({"bitpos", "a", "1", "-1", "-1"}));
+  EXPECT_EQ(24, CheckedInt({"bitpos", "a", "1", "-2", "-1"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "-1", "-2"}));
+
+  // Find set bits, explicitly mention "BYTE"
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "0", "0", "BYTE"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "0", "1", "BYTE"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0", "3", "BYTE"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0", "100", "BYTE"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "2", "2", "BYTE"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "2", "3", "BYTE"}));
+  EXPECT_EQ(32, CheckedInt({"bitpos", "a", "1", "-1", "-1", "BYTE"}));
+  EXPECT_EQ(24, CheckedInt({"bitpos", "a", "1", "-2", "-1", "BYTE"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "-1", "-2", "BYTE"}));
+
+  // Find set bits using "BIT"
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "0", "0", "BIT"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "0", "1", "BIT"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0", "21", "BIT"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "21", "21", "BIT"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "21", "100", "BIT"}));
+  EXPECT_EQ(21, CheckedInt({"bitpos", "a", "1", "0", "100", "BIT"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "-1", "-1", "BIT"}));
+  EXPECT_EQ(-1, CheckedInt({"bitpos", "a", "1", "-4", "-1", "BIT"}));
+  EXPECT_EQ(35, CheckedInt({"bitpos", "a", "1", "-5", "-1", "BIT"}));
+  EXPECT_EQ(34, CheckedInt({"bitpos", "a", "1", "-6", "-1", "BIT"}));
+
+  // Test weird return value spec, specifically when looking for clear bits in
```

Comment:
// Edge cases for for clear bits in an all-set string.

Reply:
It's actually not the edge case that I'm testing, but rather that we behave like Redis in terms of the weird spec.
I updated the comment.